src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2476,6 +2476,18 @@ void smt2_convt::convert_expr(const exprt &expr)
       out << ')';
     }
   }
+  else if(expr.id() == ID_old)
```

Comment:
We can't detect this during type checking as we can't distinguish expressions in loop invariants and in `__CPROVER_assert` at that stage.

CBMC warned that it would ignore `__CPROVER_loop_entry` in assertions during prop_conv when using SAT backend. However, when using SMT backend, CBMC would throw exception without notifying that the cause is invalid usage of loop_entry. 

This PR  patches the case of using SMT backend and clarify the correct usage of loop_entry when exception thrown.